src/openrct2/ride/Track.cpp

```diff
@@ -778,3 +779,48 @@ std::optional<CoordsXYZD> GetTrackSegmentOrigin(const CoordsXYE& posEl)
 
     return CoordsXYZD(coords, direction);
 }
+
+// Extracted from the calculation in Vehicle::UpdateMeasurements()
+bool TrackGetIsSheltered(const CoordsXYZ& input)
+{
+    // Set tile_element to first element. Since elements aren't always ordered by base height,
+    // we must start at the first element and iterate through each tile element.
+    auto tileElement = MapGetFirstElementAt(input);
+    if (tileElement == nullptr)
+        return false;
+
+    bool coverFound = false;
+    do
+    {
+        // If the tile_element is lower than the vehicle, continue (don't set flag)
+        if (tileElement->GetBaseZ() <= input.z)
+            continue;
+
+        if (tileElement->GetType() == TileElementType::Surface)
+        {
+            coverFound = true;
+            break;
+        }
+
+        if (tileElement->GetType() == TileElementType::LargeScenery)
+        {
+            coverFound = true;
+            break;
+        }
+
+        if (tileElement->GetType() == TileElementType::Path)
+        {
+            coverFound = true;
+            break;
+        }
+
+        if (tileElement->GetType() == TileElementType::SmallScenery && (tileElement->Flags & SMALL_SCENERY_FLAG_FULL_TILE) != 0)
```

Comment:
`SMALL_SCENERY_FLAG_FULL_TILE` is not a tile element flag. The original code got the scenery entry and checked the flag on that entry.